integration/spark/src/main/java/org/apache/carbondata/spark/load/CarbonLoaderUtil.java

```diff
@@ -215,6 +227,105 @@ public static void executeGraph(CarbonLoadModel loadModel, String storeLocation,
+    }
+    CarbonDataProcessorUtil
+        .validateHeader(loadModel.getTableName(), csvHeader, loadModel.getCarbonDataLoadSchema(),
+            loadModel.getCsvDelimiter(), csvFileName);
+
+    configuration.setPartitionId(loadModel.getPartitionId());
+    configuration.setSegmentId(loadModel.getSegmentId());
+    configuration.setTaskNo(loadModel.getTaskNo());
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.COMPLEX_DELIMITERS,
+        new String[] { loadModel.getComplexDelimiterLevel1(),
+            loadModel.getComplexDelimiterLevel2() });
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.SERIALIZATION_NULL_FORMAT,
+        loadModel.getSerializationNullFormat().split(",")[1]);
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.FACT_TIME_STAMP,
+        loadModel.getFactTimeStamp());
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.BAD_RECORDS_LOGGER_ENABLE,
+        loadModel.getBadRecordsLoggerEnable().split(",")[1]);
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.BAD_RECORDS_LOGGER_ACTION,
+        loadModel.getBadRecordsAction().split(",")[1]);
+    configuration.setDataLoadProperty(DataLoadProcessorConstants.FACT_FILE_PATH,
+        loadModel.getFactFilePath());
+    List<CarbonDimension> dimensions =
+        carbonTable.getDimensionByTableName(carbonTable.getFactTableName());
+    List<CarbonMeasure> measures =
+        carbonTable.getMeasureByTableName(carbonTable.getFactTableName());
+    Map<String, String> dateFormatMap =
+        CarbonDataProcessorUtil.getDateFormatMap(loadModel.getDateFormat());
+    List<DataField> dataFields = new ArrayList<>();
+    List<DataField> complexDataFields = new ArrayList<>();
+
+    // First add dictionary and non dictionary dimensions because these are part of mdk key.
+    // And then add complex data types and measures.
+    for (CarbonColumn column : dimensions) {
+      DataField dataField = new DataField(column);
+      dataField.setDateFormat(dateFormatMap.get(column.getColName()));
+      if (column.isComplex()) {
+        complexDataFields.add(dataField);
+      } else {
+        dataFields.add(dataField);
+      }
+    }
+    dataFields.addAll(complexDataFields);
+    for (CarbonColumn column : measures) {
+      // This dummy measure is added when no measure was present. We no need to load it.
+      if (!(column.getColName().equals("default_dummy_measure"))) {
+        dataFields.add(new DataField(column));
+      }
+    }
+    Iterator[] iterators = new RecordReaderIterator[recordReaders.length];
+    configuration.setDataFields(dataFields.toArray(new DataField[dataFields.size()]));
+    for (int j = 0; j < recordReaders.length; j++) {
+      iterators[j] = new RecordReaderIterator(recordReaders[j]);
+    }
+    AbstractDataLoadProcessorStep loadProcessorStep =
+        new DataLoadProcessBuilder().build(configuration, iterators);
```

Comment:
I think it is cleaner that we move all the building logic into `build` function, including the creation of recordReaders and iterators. So, for this builder, it depends on `BlockDetails` arrays which is CSV FileSplit to build the pipeline, later, we can add another builder which read from SparkRow.   
To make it more readable, we can have two utility function, one is `loadFromCSVBlocks` which accept `BlockDetails` array, and another is `loadFromIterator` which accept an iterator. (These two are out of carbon-spark package). Later, we can add another utility function call `loadFromSparkDataFrame` (probably in carbon-spark package), which will call `loadFromIterator`.

How do think about this idea?